src/theme/Segment.js

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from 'react';
+import { useLocation } from '@docusaurus/router';
+
+import '@site/static/analytics.js'
+
+const Segment = () => {
+  const { pathname } = useLocation();
+  
+  useEffect(() => {
+    if (window?.analytics) {
+      window.analytics.track('Page Viewed', {
+        page: window.location.pathname,
+        domain: window.location.origin,
+        subject_name: 'Sauce Docs',
+        product_area: 'Docs',
+        product_sub_area: 'Sauce Docs',
+      });
+    }
+  }, [pathname]);
+
```

Comment:
What if we want to do other events tracking, e.g. `Object Clicked`?

Reply:
The scope of this ticket is just to do page views, for now

Reply:
Add an `onClick` event handler and call the appropriate event